app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelTop.tsx

```diff
@@ -24,37 +24,18 @@ const connectedStateColors: Record<ConnectedState, { textColor: string; backgrou
 	[ConnectedState.UPDATE_AVAILABLE]: { textColor: '#0054cb', backgroundColor: '#e8f2ff' }
 };
 
-const setConnectedState = (
-	server: JenkinsServer,
-	isUpdatingServer: boolean,
-	updatedServer?: JenkinsServer
-): ConnectedState => {
-	let connectedState;
-
-	if (isUpdatingServer || !updatedServer) {
-		connectedState = server.connectedState;
-	} else if (updatedServer && updatedServer.uuid === server.uuid) {
-		connectedState = updatedServer?.connectedState;
-	}
-
-	return connectedState || ConnectedState.PENDING;
-};
-
 type ConnectionPanelTopProps = {
 	server: JenkinsServer,
 	refreshServers(serverToRemove: JenkinsServer): void,
 	updatedServer?: JenkinsServer,
 	isUpdatingServer: boolean
 };
 const ConnectionPanelTop = ({
-
 	server,
-	refreshServers,
-	updatedServer,
-	isUpdatingServer
+	refreshServers
 }: ConnectionPanelTopProps): JSX.Element => {
 	const history = useHistory();
-	const connectedState = setConnectedState(server, isUpdatingServer, updatedServer);
```

Comment:
This was resulting in a weird bug when refreshing. The updated version resolved the issue I was seeing.